v2/api/dbforpostgresql/customizations/flexible_server_extensions.go

```diff
@@ -79,3 +84,42 @@ func secretsToWrite(obj *postgresql.FlexibleServer) ([]*v1.Secret, error) {
 
 	return collector.Values()
 }
+
+var _ extensions.PreReconciliationChecker = &FlexibleServerExtension{}
+
+var blockingFlexibleServerStates = set.Make(
+	"starting",
+	"stopping",
+	"updating",
+)
+
+func (ext *FlexibleServerExtension) PreReconcileCheck(
```

Comment:
What's the reasoning behind having this check for PostgreSQL FlexibleServer specifically? Is it just showcasing the extension? Did we have a bug where somebody had an issue with the behavior we had previously (is there a corresponding issue?)

Generally speaking the only issue we've had that I'm aware of with blindly PUT-ing and backing off if the resource is in an `-ing` state is with AKS AgentPool, when:
1. Our backoff was to 1m, rather than 3m like now (not a ton of difference I know, but it's something).
2. The AKS AgentPool was stuck in the `-ing` state and would never leave, causing the 1m requests to continue forever. (RP bug).
3. PUT-ing to AKS AgentPool triggers AKS RP to make expensive calls to the Compute RP, which triggered compute throttling. This is also an RP bug.

I like the existence of this extension as a place where we could do GET + diff manually (if we had to, before we have it automated), or if we find a resource that specifically needs the optimization, but I'm not sure what the specific need for it here is?

If there is a specific need - can we leave a comment explaining what it is?

Reply:
On the other hand, if there is no specific need, why bother implementing this extension now? Just leave the hook for if we need it in the future?

Reply:
We've had a bug reported for PostgreSQL Flexible server, though I don't recall the exact details and can't find them in my email. But, that's why it's listed in [ADR-2022-12-Reconciliation-Extensions.md](https://github.com/Azure/azure-service-operator/blob/main/docs/hugo/content/design/ADR-2022-12-Reconciliation-Extensions.md) as an affected resource.

Reply:
It feels like we should have some details left in the code as breadcrumbs for "why we bothered to do this for this resource" - for example in this case is the problem due to a bug or bad behavior on the server side? Have we reported it to the PostgreSQL team if so? 

Do you recall how this issue was reported? Through email or otherwise? I can also try to check mine and see if I find anything, if I know more about what I'm looking for.

Reply:
I guess the issue is here #2643